### ISSUE TITLE
Wallpaper Engine

### ISSUE THREAD

**Issue by James-Slam** [mon/0x40-web#28]
Hiya,

I'm trying to get your version of 0x40 hues to work with Wallpaper Engine: http://steamcommunity.com/app/431960 

It has a feature that lets you set a webpage to your desktop. The problem is that it errors when trying to fetch the respacks with error 0. I tried calling the page from another html file so that I could invoke query strings, but ran into the same issue. What does seem to work is adding respacks through the built in options menu, as long as you call none at the start. 

Is there any easy way to call the function that imports local zips from the option menu? If so, would I be able to just call that function on load to fetch after the page is initialized automatically? 

Thanks,
James

**Comment by mon** [comment]
Can I get a screenshot of the exact error? I have a hunch about the way they're implementing the web view, if it's correct then it will be very difficult to make it work.

**Comment by James-Slam** [comment]
![capture](https://cloud.githubusercontent.com/assets/3708601/22817949/5eb41248-ef38-11e6-9aec-f953d56cfca8.PNG)

**Comment by James-Slam** [comment]
I tried calling it from the absolute path too, but ran into the same issue.

**Comment by mon** [comment]
I'm reading the communities and it says when you create the wallpaper package you need to select ALL files required, definitely grabbing the `respacks` folder as well as DefaultsHQ.zip ?

**Comment by James-Slam** [comment]
I was just pointing at the index.html file before. I figured it could see the other files since there were no errors and my added css was working.

I made a new project this time instead of just pointing at a file.
![capture](https://cloud.githubusercontent.com/assets/3708601/22829566/33ba3624-ef71-11e6-97d6-e343e2d61a39.PNG)

When I create the project, it doesn't keep the respacks directory imported:
![capture3](https://cloud.githubusercontent.com/assets/3708601/22829593/55af3a0e-ef71-11e6-9eba-393128cd874f.PNG)

I tried including the repacks in the root folder and the application saw them, but still gave the same error when I called them. After testing, I found out that if I include a blank html file in the respacks directory, it will import the respack zips. Still gives the same error though.
![capture4](https://cloud.githubusercontent.com/assets/3708601/22829889/8f4976ca-ef72-11e6-8908-50e045154a98.PNG)

Thanks for helping me with this!
James

**Comment by mon** [comment]
To me it looks like it's still having issues actually finding the files, and the fact they were silently excluded when there was no html file is dodgy.

If you'd like, you can try to do a dirty hack... Change the extension from `.zip` to `.png`, and edit `index.html` to reference the respacks by that new extension. Try not to accidentally open the png, as that obviously won't work ;) Perhaps the "accepted" extension could convince it to bundle the files properly.

**Comment by James-Slam** [comment]
Yeah I found it weird that it silently excluded the whole directory when it only had .zip files too.

I tried renaming DefaultsHQ.zip to a png and it worked in Firefox, but not Wallpaper Engine :/ 

![capture](https://cloud.githubusercontent.com/assets/3708601/22838677/48eb8b74-ef94-11e6-8af1-4d7cd8a5fd29.PNG)

**Comment by mon** [comment]
It could be a CORS issue - I know that Chrome can't just open the HTML, you need to be running a web server for it to pass origin checks when loading the zips.

Is there any way to get a console? I'd like to see if there's anything showing up there that I'm not displaying.

You may need to move this to the dev of Wallpaper Engine - he knows more about his system than me ;)

**Comment by mon** [comment]
Long time, I know... Did you manage to work this out?

**Comment by James-Slam** [comment]
Hi mon, 

Sorry for not following up, I didn't manage to fix the problem.

I only occasionally 0x40 hues as a wallpaper, so the work around by loading character and audio packs through the settings is my go to. Auto running would be nice still though if you managed to figure it out. 

Thanks for your help before!

James

**Comment by mon** [comment]
No worries, I'll keep the issue open if anyone ever works it out 👍 

It could potentially be fixed by embedding the respack as a base64 string, but it would be pretty filthy.

**Comment by scillidan** [comment]
I don't know how to describe the principle, but you need to run `serve` in the folder.
In folder of `0x40-web`, `npm i -g serve`, then `serve  .`, then open address that should be `localhost:xxxx`. The `index.html` should start to work.
Or use [miniserve](https://github.com/svenstaro/miniserve) or [dufs](https://github.com/sigoden/dufs).

Thanks mon 🍻, he solved the thorny problem of CROS. Then, all you have to do is click `Settings`, click `Resources` and
 download the `remote respacks`.

Finally, I secretly 👻 uploaded `0x40-web` to Wallpaper Engine. Here is https://steamcommunity.com/sharedfiles/filedetails/?id=2888591333
